### ISSUE TITLE
Hierarchical repository ownership for organizations

### ISSUE THREAD

**Issue by harrism** [isaacs/github#49]
At my company, we have a number of open-source projects on github, and they are spread out in various user and organization accounts.  We are considering consolidating under one big github organization, but that has a big problem: when we do that, the organization will have one flat, monolithic repository list and it will be hard for the public (and us) to navigate it.  And individual groups within our company won’t get recognition for their own repositories.

It would be great if github could add the ability to have hierarchy in an organization’s repositories, without git itself having to be aware of that hierarchy.  A possible alternative is to allow organizations to be members of other organizations, and display the hierarchy of repositories below each organization on that org’s github page.

This would be a big help for large organizations like ours.

**Comment by sbrossie** [comment]
:+1:

**Comment by ericvh** [comment]
+1

**Comment by mccreigh** [comment]
+1

**Comment by harrism** [comment]
See also issue #136.

**Comment by bliu033** [comment]
+1

**Comment by gouldner** [comment]
+1

**Comment by github-s** [comment]
+1

**Comment by russtoku** [comment]
+1

**Comment by jdbeutel** [comment]
+1

**Comment by github-s** [comment]
+1

**Comment by bkfitzwa** [comment]
:+1:

**Comment by franklinwise** [comment]
+1

**Comment by kamar535** [comment]
+1

**Comment by phipleg** [comment]
+1

**Comment by tikenn** [comment]
+1

**Comment by loneill7** [comment]
We will have the same issue. Across our organisation we could have a few dozen projects on the go at any one time plus probably 10 times more historic projects. With one repository per project, it gets unwieldly very quickly. We plan to address this with a naming convention for repositories [unit]-[projType]-[projectName] but it is not ideal.  The ability to tag repositories would help us out here too.

**Comment by jlutzhbo** [comment]
Yes please!

**Comment by praskris** [comment]
we would love to have this feature as well. This is essential when you have business orgs and several teams. The ability to have that hierarchy would be valuable.

**Comment by superherointj** [comment]
+1

**Comment by willjeong** [comment]
+1

**Comment by tomdurst** [comment]
+1

**Comment by joeherwig** [comment]
+1

**Comment by zsoltbalint** [comment]
+1

**Comment by markus-albis** [comment]
+1

**Comment by sumanmehta** [comment]
+1

**Comment by patrickleet** [comment]
+1

**Comment by darshakthakore** [comment]
+1

**Comment by AlecBenzer** [comment]
Please don't reply to an issue just to say '+1': it sends a notification to everyone subscribed to the issue. Instead, just react to the issue's initial comment with the 👍 emoji.


<img width="233" alt="plus_1_github" src="https://user-images.githubusercontent.com/92369/31297680-0455e07c-aab5-11e7-81dc-d6070ca75378.png">

**Comment by sappy5678** [comment]
+1

**Comment by npopov-d3** [comment]
+1

**Comment by metastableB** [comment]
+1

**Comment by msdewitt** [comment]
+1

**Comment by carterbox** [comment]
I believe the features of this request are all addressed by GitHub teams. Within a single organization you can create teams of users who each get their own [team pages](https://help.github.com/en/articles/about-teams#team-pages). Furthermore, you can have nested teams.

**Comment by dmiluski** [comment]
Hi @carterbox ,

Although Teams supports the tree/nested structures, one pain point I've found is that I'd love to setup a wiki per "Team". This could help for onboarding document handling when focused on a "vertical team" (some function), as well as "horizontal team" (some platform).

Would you have suggestions on how to link those? Potentially separate repos which have their own readme/wiki?

**Comment by carterbox** [comment]
@dmiluski. Exactly, only projects have wikis, so just have each team set up an empty repository from their team page (teams can own repositories).

**Comment by davidnice-ONE** [comment]
This has clearly been a problem for some time!   While I'd agree that teams and sub-teams help, I think that the lack of a 'dashboard context' for teams is a show-stopper.

I work for a large organisation with hundreds of users and repos, but I can't just see my team on the dashboard.  I have to click through five pages to get to the team page, and then the repo list still includes repos that have been shared with us so there's still no way to view just my teams repos, issues and activity like there would be on a dashboard.

In short, how about a dashboard context for teams?  Does that sound like a useful idea?

**Comment by dmiluski** [comment]
Since this post, I have since moved from vanilla Github to Zenhub which tackles a number of my requirements by organizing and linking tasks between platform repos that can be viewed/organized on a single team dashboard.
